src/tools/rbd/action/Wnbd.cc

```diff
@@ -172,29 +128,26 @@ int execute_unmap(const po::variables_map &vm,
   std::cerr << "rbd: wnbd is only supported on Windows" << std::endl;
   return -EOPNOTSUPP;
 #else
-  std::string device_name = utils::get_positional_argument(vm, 0);
-
   std::string image_name;
-  if (device_name.empty()) {
-    int r = get_image_or_snap_spec(vm, &image_name);
-    if (r < 0) {
-      return r;
-    }
+
+  int r = utils::get_image_or_snap_spec(vm, &image_name);
+  if (r < 0) {
+    return r;
   }
 
-  if (device_name.empty() && image_name.empty()) {
-    std::cerr << "rbd: unmap requires either image name or device path"
+  if (image_name.empty()) {
```

Comment:
I don't think this check is needed now as for wnbd unmap is basically the same as map when it comes to these options.  And map doesn't have such a check in any of the device types.